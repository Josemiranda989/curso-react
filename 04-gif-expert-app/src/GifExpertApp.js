// rafc > crea un componente

import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  /*     const handleAdd = () => {
        //setCategories([...categories, 'Dragon Ball'])
        setCategories(cat => [...cat, 'Dragon Ball'])
    } */

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      {/* <AddCategory setCategories={setCategories} categories={categories} /> Metodo2 */}
      <hr />

      <ol>
        {categories.map((category, i) => {
          return <GifGrid category={category} key={category + i} />;
        })}
      </ol>
    </>
  );
};
