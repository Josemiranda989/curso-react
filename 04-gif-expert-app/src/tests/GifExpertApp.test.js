
import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en <GifExpertApp/>', () => {

test('debe mostrar correctamente el componente', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
});
    
    test('debe de mostrar una lsita de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    });


})