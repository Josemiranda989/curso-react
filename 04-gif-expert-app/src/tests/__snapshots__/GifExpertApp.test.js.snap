// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp/> debe de mostrar una lsita de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch0"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball1"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp/> debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
